include/signalrclient/signalr_value.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace signalr
+{
+    class value
+    {
+    public:
+        enum type
+        {
+            MAP,
+            ARRAY,
+            STRING,
+            NUMBER,
+            NULL,
+            BOOL
+        };
+
+        value();
+        value(int val);
+        value(double val);
+        value(const std::string& val);
+        value(const std::vector<value>& val);
+        value(const std::string& property_name, int val);
+        value(const std::string& property_name, double val);
+        value(const std::string& property_name, const std::string& val);
+        value(const std::string& property_name, const std::vector<value>& val);
+
+        bool is_map() const;
```

Comment:
Do we have doc comments support at all? I'd like to fill out the doc comments here before finalizing APIs so we have some initial documentation content in place.

Reply:
I haven't seen any evidence of C++ docs through our doc system. It looks like the azure storage cpp sdk uses doxygen and hosts their own GitHub page with the generated docs.

Reply:
Can we throw some doxygen in then? Maybe ask around the docs folks and see what (if any) options we have.